### PR TITLE
Raise a more appropriate exception when unable to parse a file

### PR DESCRIPTION
**Pull request by jsantos17**
Depends on: slamdata/quasar#3776

The exception now includes a hint towards a misconfigured datasource when unable to parse a resource.

The quasar-version in this PR needs to bumped after slamdata/quasar#3776 is merged.

### CODE REVIEW

**Comment by jsantos17** [comment]
Breaking since this makes `quasar-s3` incompatible with older versions of Quasar.

**Comment by jsantos17** [comment]
Now the message in the responses looks like:

```
{
  "detail": {
    "path": "/post-giraffe.ldjson",
    "message": "Unable to open resource: Could not parse the file as JSON. Ensure you've configured the correct jsonParsing option for this bucket"
  },
  "status": "Bad Request",
  "name": "malformed-resource"
}
```

I used 400 Bad Request for the status code, but not sure if that's appropriate. And of course the details in the error message are controlled by the datasource, not Quasar.

cc @beckyconning

File: datasource/src/main/scala/quasar/physical/s3/impl/evaluate.scala [hunk 1]
```diff
@@ -93,17 +103,24 @@ object evaluate {
...
-        // and we're done.
-        asData
-      })
+          // and we're done.
+          asData
+        })
+
+      stream.map(_.handleErrorWith {
+        case ParsingFailure(_, _) =>
+          Stream.raiseError(QuasarError.throwableP(noParseError(file)))
```
Comment (rintcius) [inline datasource/src/main/scala/quasar/physical/s3/impl/evaluate.scala]: I think we shouldn't lose the exact `ParsingFailure` that happened (message and underlying cause). I.e. this info should be supplied as part of `noParseError` in some way (together with the expected format)).

Comment (jsantos17) [reply datasource/src/main/scala/quasar/physical/s3/impl/evaluate.scala]: 👍

Comment (jsantos17) [reply datasource/src/main/scala/quasar/physical/s3/impl/evaluate.scala]: I now retain the `message` from `ParsingFailure`. The message from the `underlying` is exactly the same as the message for this particular exception.

**Comment by jsantos17** [comment]
@wemrysi 

>Not that it need be done in this PR, but we may want to perform a HEAD request on a resource and look at the Content-Type before attempting to parse. If it is a type that we know we can't support then we can just error with something more specific

We already do a version of this to determine if a resource exists or if it's merely a prefix. Might be easy enough to improve it to check the `Content-Type` too.

**Comment by rintcius** [comment]
@jsantos17 ready to bump version - feel free to merge after that when it's green

**Comment by jsantos17** [comment]
@rintcius @wemrysi Bumped `quasar-version` and build should be green, but now there's some sort of binary incompatibility with the last version of quasar, scalaz and this repo. Investigating...

**Comment by jsantos17** [comment]
Also, this incompatibility is why we have https://app.clubhouse.io/data/story/565/verify-quasar-s3-loads-into-slamdata-backend-in-the-travis-build

**Pull request merged**